doc: nrf: Add a section about hostap (#18351)
nrfconnect/sdk-nrf: doc/nrf/app_dev/device_guides/nrf70/features.rst

Peer-to-peer support in the form of Wi-Fi Direct® will be available in the future.

See the :ref:`ug_wifi` documentation for more information related to Wi-Fi modes of operation.

hostap
******

The nRF70 Series devices use the `WPA Supplicant`_ to implement full Wi-Fi functionality.
The WPA supplicant is part of the ``hostap`` project and is a widely used implementation of the IEEE 802.11i standard for wireless LAN security.

> The WPA supplicant is a software component that implements the WPA, WPA2 and WPA3 security protocols.

Reviewer: ```suggestion
The WPA supplicant is a software component that implements the Wi-Fi Protected Access (WPA™), WPA2™, and WPA3™ security protocols.
```

> The nRF70 Series devices use `Zephyr hostap fork`_ a fork of the ``hostap`` project that is integrated with the Zephyr RTOS.

Reviewer: ```suggestion
The nRF70 Series devices use `Zephyr hostap fork`_ , a fork of the hostap project that is integrated with the Zephyr RTOS.
```

> The WPA supplicant is integrated with the Zephyr RTOS and registers as a Wi-Fi network manager in the Zephyr networking stack, see `Zephyr Wi-Fi NM API`_ for details.

Reviewer: ```suggestion
The WPA supplicant is integrated with the Zephyr RTOS and registers as a Wi-Fi network manager in the Zephyr networking stack.
See `Zephyr Wi-Fi NM API`_ for details.
```

The `Zephyr Wi-Fi management`_ layer in Zephyr uses the Wi-Fi network manager to manage the Wi-Fi interface.

> The nRF70 Series drivers registers as a Wi-Fi device in the Zephyr networking stack and provides the Wi-Fi interface to the WPA supplicant.

Reviewer: ```suggestion
The nRF70 Series drivers register as a Wi-Fi device in the Zephyr networking stack and provide the Wi-Fi interface to the WPA supplicant.
```

Author: Modified from drivers->driver and didn't do the other changes

The WPA supplicant then manages the Wi-Fi interface and provides the Wi-Fi functionality to the application.

.. note::

>       The WPA supplicant is only used for ``system mode`` to offer full Wi-Fi functionality.
>       It is not used in other modes, for example ``scan_only`` or ``monitor`` modes.

Reviewer: ```suggestion
      The WPA supplicant is only used for System mode to offer full Wi-Fi functionality.
      It is not used in other modes, for example, Scan-only or Monitor modes.
```

NCS scope

> =========

Reviewer: Can the title be changed to- "Supported hostap features in the |NCS|"??

Peer-to-peer support in the form of Wi-Fi Direct® will be available in the future.

See the :ref:`ug_wifi` documentation for more information related to Wi-Fi modes of operation.

> 

Reviewer: This label could be added to reference it in the Changelog.
```suggestion
.. _ug_nrf70_features_hostap:

```
